### PR TITLE
Fixes from YAML file testing

### PR DESCRIPTION
**Pull request by yahskapar**
- [x] PURE_SIGNAL
- [x] PURE_PURE_UBFC_TSCAN (Pre-process training, validation, and test, USE_LAST_EPOCH: False)
- [x] PURE_PURE_UBFC_TSCAN (Pre-process training and test, USE_LAST_EPOCH: True)
- [x] PURE_PURE_UBFC_PHYSNET (Pre-process training and test, USE_LAST_EPOCH: True)
- [x] PURE_PURE_UBFC_EFFICIENTPHYS (Pre-process training and test, USE_LAST_EPOCH: True)
- [x] PURE_PURE_UBFC_DEEPPHYS (Re-use pre-processed datasets from TSCAN YAML, USE_LAST_EPOCH: True)
- [x] PURE_PURE_UBFC_DEEPPHYS (Re-use pre-processed datasets from TSCAN YAML, USE_LAST_EPOCH: False)
- [x] UBFC_UBFC_PURE_TSCAN (Pre-process training and test, USE_LAST_EPOCH: True)
- [x] UBFC_UBFC_PURE_DEEPPHYS (Re-use pre-processed datasets from TSCAN YAML, USE_LAST_EPOCH: True)

All of the above experiments were done with 4 GPUs instead of 1 (the default).

---Results---

* PURE SIGNAL (POS)
    * Peak MAE (Peak Label):7.979154190236877
    * PEAK RMSE (Peak Label):11.18274213693747 
    * PEAK MAPE (Peak Label):10.76341058013474 
    * PEAK Pearson  (Peak Label):0.8930326227658032
* PURE_PURE_UBFC_TSCAN (Pre-process training and test, USE_LAST_EPOCH: True)
    * FFT MAE (FFT Label):1.2346540178571428
    * FFT RMSE (FFT Label):2.854432455289637
    * FFT MAPE (FFT Label):1.4367665761122275
    * FFT Pearson (FFT Label):0.9889652657707192
* PURE_PURE_UBFC_TSCAN (Pre-process training, validation, and test, USE_LAST_EPOCH: False)
    * FFT MAE (FFT Label):1.2346540178571428
    * FFT RMSE (FFT Label):2.854432455289637
    * FFT MAPE (FFT Label):1.4367665761122275
    * FFT Pearson (FFT Label):0.9889652657707192
* PURE_PURE_UBFC_PHYSNET (Pre-process training and test, USE_LAST_EPOCH: True)
    * FFT MAE (FFT Label):3.3272879464285716
    * FFT RMSE (FFT Label):7.798929690192735
    * FFT MAPE (FFT Label):3.224378158400154
    * FFT Pearson (FFT Label):0.9112985647107084
* PURE_PURE_UBFC_EFFICIENTPHYS (Pre-process training and test, USE_LAST_EPOCH: True)
    * FFT MAE (FFT Label):3.6411830357142856
    * FFT RMSE (FFT Label):9.624114213429523
    * FFT MAPE (FFT Label):3.475852242974659
    * FFT Pearson (FFT Label):0.860732821226475
* PURE_PURE_UBFC_DEEPPHYS (Re-use pre-processed datasets from TSCAN YAML, USE_LAST_EPOCH: True)
    * FFT MAE (FFT Label):1.0672433035714286
    * FFT RMSE (FFT Label):2.802411103961278
    * FFT MAPE (FFT Label):1.2295848620073073
    * FFT Pearson (FFT Label):0.9887672305096127
* PURE_PURE_UBFC_DEEPPHYS (Re-use pre-processed datasets from TSCAN YAML, USE_LAST_EPOCH: False)
    * FFT MAE (FFT Label):1.0672433035714286
    * FFT RMSE (FFT Label):2.802411103961278
    * FFT MAPE (FFT Label):1.2295848620073073
    * FFT Pearson (FFT Label):0.9887672305096127
* UBFC_UBFC_PURE_TSCAN (Pre-process training and test, USE_LAST_EPOCH: True)
    * FFT MAE (FFT Label):5.758351293103448
    * FFT RMSE (FFT Label):17.777401220875497
    * FFT MAPE (FFT Label):6.220913030646447
    * FFT Pearson (FFT Label):0.6882068130947284
* UBFC_UBFC_PURE_DEEPPHYS (Re-use pre-processed datasets from TSCAN YAML, USE_LAST_EPOCH: True)
    * FFT MAE (FFT Label):5.796235183189655
    * FFT RMSE (FFT Label):18.848054277457575
    * FFT MAPE (FFT Label):5.744317173377135
    * FFT Pearson (FFT Label):0.651010524782676

### CODE REVIEW

**Comment by yahskapar** [comment]
TODO: Look into discrepancy in results between USE_LAST_EPOCH: True and USE_LAST_EPOCH: False with PURE_PURE_UBFC_TSCAN. It's not clear why these results are different at all since the only difference should be how the epoch used for testing is selected rather than anything to do with the training data.

**Comment by yahskapar** [comment]
> TODO: Look into discrepancy in results between USE_LAST_EPOCH: True and USE_LAST_EPOCH: False with PURE_PURE_UBFC_TSCAN. It's not clear why these results are different at all since the only difference should be how the epoch used for testing is selected rather than anything to do with the training data. In both cases, despite the results being different, the last epoch is used or selected anyhow.

@xliucs @girishvn I fixed this after consulting this relevant issue: https://github.com/pytorch/pytorch/issues/80119#issuecomment-1164630341.

The problem is that all of the torch objects, such as the dataloaders, share the same generator with the same fixed seed. At the very least, the training dataloader's seed should differ from the rest of the objects's seed so that the training dataloader can be isolated and controlled for randomness. The training results are consistent, regardless of the validation dataset being used or not, after adding this separate, fixed seed for the training dataloader.

**Review by xliucs — APPROVED** [review]
